readme.md

```diff
@@ -10,248 +8,408 @@ See also: https://github.com/ga-wdi-lessons/ruby-oop
+
+  ```rb
+  adrian = Person.new("Adrian", 29)
+  ```
+</details>
+<br/>
+<details>
+  <summary><strong>How do we go about accessing/modifying a <code>Person</code> instance's <code>name</code> and <code>age</code> attributes?</strong></summary>
 
-  def demonstrate_getting_name
-    # goal is to get our person's name in the best way possible
-    # GETTERS
-    puts @name
-    puts name
-    puts self.name
+  ```rb
+  adrian.name
+  # => "Adrian"
+  adrian.name = "Dwayne 'The Rock' Johnson"
+  adrian.name
+  # =>"Dwayne 'The Rock' Johnson"
+
+  adrian.age
+  # => 29
+  adrian.age = 1
+  # => undefined method `age='
+  ```
+
+</details>
+
+--------
+
+<details>
+  <summary><strong>Let's give our <code>Person</code> class a <code>say_name</code> method that prints his/her name to the console.</strong></summary>
+
+  ```rb
+  class Person
+    attr_accessor :name
+    attr_reader :age
+
+    def initialize(initial_name, initial_age)
+      @name = initial_name
+      @age = initial_age
+    end
+
+    def say_name
+      puts "Hi, my name is " + @name + "."
+    end
   end
 
-  def demonstrate_setting_hunger_level
-    # goal is to set/update our person's hunger level in the best way possible
-    # SETTERS
-    @hunger_level = -10
-    hunger_level = -10
-    self.hunger_level = -10
+  adrian = Person.new("Adrian", 29)
+  adrian.say_name
+  # => "Hi, my name is Adrian."
+  ```
+
+  > We could also use `self.name` in place of the instance variable `@name`. This is actually preferable. We'll explain why later...
+
+</details>
+
+<details>
+  <summary><strong>What type of method is <code>say_name</code>? What other types of methods are there?</strong></summary>
+
+  `say_name` is an instance method. It is a method that we would call on an instance of `Person`.
+
+  Ruby classes can define two types of methods...
+  * **Instance:** called on on single instances.
+  * **Class:** called on the class, deal with the set of objects.
+
+  Instance and Class methods are both common and okay to use. This is unlike class variables, which should be used sparingly.
+
+</details>
+
+--------
+
+<details>
+  <summary><strong>Now we want to keep track of how many people have been created using the <code>Person</code> class. What tools can we use to do this? How would we use them?</strong></summary>
+
+  ```rb
+  class Person
+    attr_accessor :name
+    attr_reader :age
+    # We create a class variable that is an array contains all existing instances of `Person`
+    @@people = []
+
+    def initialize(initial_name, initial_age)
+      @name = initial_name
+      @age = initial_age
+      # Whenever we create a new person, we push it into the `@@people` array
+      @@people.push(self)
+    end
+
+    def say_name
+      puts "Hi, my name is " + @name + "."
+    end
+
+    # Here we're creating a method that retrieves the value of `@@people`
+    def Person.get_people
```

Comment:
Was hoping to replace with `self` later when refactoring but yeah, I should just introduce it here.